File: assignments/a1/q4/Java/Goo.java
//Goo:top//
//Goo:import//
import java.util.*;

// <blah>:Goo ::= THIS <VAR> IS <silly>
public class Goo extends Blah /*Goo:class*/ {

    public static final String $className = "Goo";
    public static final String $ruleString =
        "<blah>:Goo ::= THIS <VAR> IS <silly>";

    public Token var;
    public Silly silly;

    public Goo(Token var, Silly silly) {
//Goo:init//
        this.var = var;
        this.silly = silly;
    }

    public static Goo parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<blah>:Goo", scn$.lno);
        scn$.match(Token.Match.THIS, trace$);
        Token var = scn$.match(Token.Match.VAR, trace$);
        scn$.match(Token.Match.IS, trace$);
        Silly silly = Silly.parse(scn$, trace$);
        return new Goo(var, silly);
    }

//Goo//
}

File: assignments/a1/q4/Python/Goo.py
#Goo:top#
#Goo:import#

# <blah>:Goo ::= THIS <VAR> IS <silly>
class Goo(Blah): #Goo:class#

    className = "Goo"
    ruleString = "<blah>:Goo ::= THIS <VAR> IS <silly>"
    var = None
    silly = None

    def __init__(var, silly):
        #Goo:init#
        self.var = var
        self.silly = silly

#Goo#
